src/components/TokenNFTPreview.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState, useEffect } from 'react';
+
+const TokenNFTPreview = (props) => {
+  const [token, setToken] = useState(props.token);
+
+  useEffect(() => {
```

Comment:
Why do we need the `useEffect` hook? You could just access `props.token` directly or am I missing something?

Reply:
When token is first passed, it doesn't have all attributes set. `useEffect` is needed to update the state when `token` change:

https://stackoverflow.com/a/59289634

Reply:
I see! 

Can you just add a comment to explain that you are using the useEffect to force a render?